library.c

```diff
@@ -2266,6 +2266,11 @@ redis_sock_check_liveness(RedisSock *redis_sock)
     smart_string cmd = {0};
     size_t len;
 
+    /* Short circuit if we detect the stream is "dirty", can't or are
+       configured not to try and fix it */
+    if (redis_stream_detect_dirty(redis_sock->stream) != SUCCESS)
+        goto failure;
+
     /* Short circuit if we detect the stream has gone bad or if the user has
      * configured persistent connection "YOLO mode". */
     if (redis_stream_liveness_check(redis_sock->stream) != SUCCESS) {
```

Comment:
I'm wondering if we should reverse these checks since the `redis_stream_liveness_check` is likely very fast and if the stream isn't even live, we don't need to check if there are unconsumed bytes?